src/libsystemd-network/sd-dhcp-client.c

```diff
@@ -1684,7 +1684,9 @@ static int client_receive_message_udp(
         assert(client);
 
         buflen = next_datagram_size_fd(fd);
-        if (buflen < 0)
+        if (buflen == -ENETDOWN)
+                return 0;
```

Comment:
can you add a comment here, please, explaining the situation?

Reply:
Ok, done.